__init__.py

```diff
@@ -0,0 +1,143 @@
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from neon_utils.skills.neon_skill import NeonSkill, LOG
+from mycroft.skills.core import intent_file_handler
+from .request_handling import RequestHandler
+
+
+
+class DirectorySkill(NeonSkill):
+
+    def __init__(self):
+        super(DirectorySkill, self).__init__(name="DirectorySkill")
+
+    def initialize(self):
+        # When first run or prompt not dismissed, wait for load and prompt user
+        if self.settings.get('prompt_on_start'):
+            self.bus.once('mycroft.ready', self._start_mall_parser_prompt)
+
+    @intent_file_handler("run_mall_parser.intent")
+    def start_mall_parser_intent(self, message):
+        LOG.info(message.data)
+
+        self._start_mall_parser_prompt(message)
+        return
+
+    def mall_link(self):
+        mall_link = 'https://www.alamoanacenter.com/'
+        return self.settings.get("mall_link") or mall_link
+
+    def user_request_handling(self, message):
+        LOG.info(f"Message is {message.data}")
+        request_lang = message.data['lang'].split('-')[0]
+        user_request = message.data['shop']
+        LOG.info(f"{self.mall_link()}")
+        LOG.info(str(request_lang))
+        LOG.info(user_request)
+        found, link = RequestHandler.existing_lang_check(request_lang, self.mall_link())
+        if found:
+            link = self.mall_link()+request_lang+'/directory/'
+            LOG.info('new link: '+ link)
+            return user_request, link
+        else:
+            self.speak_dialog("no_lang")
+            return None, None
+
+    def start_again(self):
+        start_again = self.ask_yesno("ask_more")
+        if start_again == "yes":
+            another_shop = self.get_response('another_shop')
+            LOG.info('another shop'+another_shop)
+            return another_shop
+        elif start_again == "no":
+            return None
+        else:
```

Comment:
Consider handling no response as 'no' to avoid spoken errors after the skill interaction is finished; or speak some confirmation that the skill isn't waiting for a response (i.e. "If you would like to locate another shop, just ask me where it is.")